### PR TITLE
Encode int64 numbers

### PR DESCRIPTION
**Pull request by lajonat**
Hey,
I changed the int64 encoding/decoding functions to encode them like the spec defines.
This doesn't handle precision issues, as I haven't changed the type, I just changed the way the bytes are written to the output.
I've tested it, and it interoperates against the java msgpack library correctly.

### CODE REVIEW

File: lib/decoder.js [hunk 1]
```diff
@@ -142,7 +142,11 @@ module.exports = function buildDecode(decodingTypes) {
         return buildDecodeResult(result, 5)
       case 0xcf:
         // 8-bytes BE unsigned int
-        result = buf.readUInt32BE(offset + 5) * 0xffffffff + buf.readUInt32BE(offset + 1)
+		// Read long byte by byte, big-endian
+        var result = 0;
+        for (var k = 7; k >= 0; k--) {
+            result += (buf.readUInt8(offset + k + 1) * Math.pow(2 , (8 *(7-k))));
+        }
```
Comment (mcollina) [inline lib/decoder.js]: can you please define the variables at the top?

File: lib/encoder.js [hunk 2]
```diff
@@ -219,9 +219,11 @@ module.exports = function buildEncode(encodingTypes) {
...
 
 function write64BitUint(buf, obj) {
-  var big = Math.floor(obj / 0xffffffff)
-  buf.writeUInt32BE(big, 5)
-  buf.writeUInt32BE(obj - big * 0xffffffff, 1)
+   // Write long byte by byte, in big-endian order
+  for (var k = 7; k >= 0; k--) {
+        buf[k + 1] = (obj & 0xff);
+        obj = obj / 256;
+    }
```
Comment (mcollina) [inline lib/encoder.js]: The indenting seems wrong here.

File: test/64-bits-unsigned-integers.js [hunk 3]
```diff
@@ -18,7 +18,10 @@ test('encoding/decoding 64-bits big-endian unsigned integers', function(t) {
       var buf = encoder.encode(num)
       t.equal(buf.length, 9, 'must have 9 bytes')
       t.equal(buf[0], 0xcf, 'must have the proper header')
-      t.equal(buf.readUInt32BE(5) * base + buf.readUInt32BE(1), num, 'must decode correctly');
+      var result = 0; 
+      for (var k = 7; k >= 0; k--) {
+           result += (buf.readUInt8(k + 1) * Math.pow(2 , (8 *(7-k))));
+       }
```
Comment (mcollina) [inline test/64-bits-unsigned-integers.js]: You should add back the `t.equal()`.

**Comment by lajonat** [comment]
Fixed.

**Comment by mcollina** [comment]
THanks! releasing it now!

**Pull request merged**